info-compiler/src/test/java/com/google/sps/WebCrawlerTest.java

```diff
@@ -12,12 +12,215 @@
+  private final static String VALID_URL =
+    "https://www.cnn.com/2020/06/23/politics/aoc-ny-primary-14th-district/index.html";
+  private final static String VALID_URL_ROBOTS_TXT =
+    "https://www.cnn.com/robots.txt";
+  private final static NewsArticle EXPECTED_NEWS_ARTICLE =
+    new NewsArticle(
+        "AOC wins NY Democratic primary against Michelle Caruso-Cabrera, CNN projects - " +
+        "CNNPolitics",
+        VALID_URL,
+        "Washington (CNN)Freshman Democratic Rep. Alexandria Ocasio-Cortez will defeat former " +
+        "longtime CNBC correspondent and anchor Michelle Caruso-Cabrera in a Democratic " +
+        "primary election on Tuesday for New York's 14th Congressional District, CNN projects.");
+  private final static String EMPTY_ABBREVIATED_CONTENT = "";
+  private final static int DELAY = 1;
+
+  private static WebCrawler webCrawler;
+  private static LocalDatastoreHelper datastoreHelper;
+  private static Datastore datastore;
+
+  @BeforeClass
+  public static void initialize() throws InterruptedException, IOException {
+    datastoreHelper = LocalDatastoreHelper.create();
+    datastoreHelper.start();
+    datastore = datastoreHelper.getOptions().getService();
+    webCrawler = new WebCrawler(datastore);
+  }
+
+  /**
+   * Resets the internal state of the Datastore emulator and then {@code datastore}. Also resets
+   * {@code webCrawler}. We choose to reset, instead of creating/destroying the Datastore emulator
+   * at each test is because {@code datastoreHelper.stop()} sometimes generates a {@code
+   * java.net.ConnectException}, when making HTTP requests. Hence we try to limit the number of
+   * times {@code datastoreHelper} is created/destroyed.
+   */
+  @Before
+  public void resetDatastore() throws IOException {
+    datastoreHelper.reset();
+    datastore = datastoreHelper.getOptions().getService();
+    webCrawler = new WebCrawler(datastore);
+  }
+
+  // @TODO [Implement unit tests for getting URLs from the Custom Search engine.]
+  @Test
+  public void getUrlsFromCustomSearch() {}
+
+  @Test
+  public void scrapeAndExtractFromHtml_validUrl() throws IOException {
+    // Scrape and extract news article content from {@code VALID_URL}. The content and metadata
+    // packaged in {@code NewsArticle} should be consistent with that in {@code
+    // EXPECTED_NEWS_ARTICLE}.
+    // Assume that the libraries {@code URL} and {@code RobotsTxt} work as intended.
+    // Since Mockito doesn't support the mocking of static methods, {@code NewsContentExtractor}'s 
+    // {@code extractContentFromHtml()} is not insular to this "unit" test. @TODO [Might modify
+    // {@code NewsContentExtractor} to aid test-driven development.
+    URL url = new URL(VALID_URL);
+    Optional<NewsArticle> potentialNewsArticle = webCrawler.scrapeAndExtractFromHtml(url);
+    Assert.assertTrue(potentialNewsArticle.isPresent());
+    Assert.assertEquals(EXPECTED_NEWS_ARTICLE, potentialNewsArticle.get());
+  }
```

Comment:
nit: Don't test with real data.  Probably can just get rid of this test since pretty much all of the custom logic is in politelyscrapeAndExtractFromHtml anyways.